crates/holochain-conductor-runtime/src/runtime.rs

```diff
@@ -181,6 +175,33 @@ impl Runtime {
         }
     }
 
+    /// Full process to setup an app
+    ///
+    /// Check if app is installed, if not install it, then optionally enable it.
+    /// Then ensure there is an app websocket and authentication for it.
+    pub async fn setup_app(
+        &self,
+        payload: InstallAppPayload,
+        enable_after_install: bool,
+    ) -> RuntimeResult<AppAuth> {
+        // This is a temporary workaround because we cannot clone AppBundleSource,
+        // which is needed to read the actual app name from the manifest
+        // See https://github.com/holochain/holochain/pull/4882
+        let installed_app_id = payload
+            .installed_app_id
+            .clone()
+            .ok_or(RuntimeError::InstalledAppIdNotSpecified)?;
```

Comment:
Rather than deal with making a new type, I'm just erroring if the caller doesn't provide an installed_app_id. This is temporary workaround.